jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java

```diff
@@ -18,15 +18,171 @@
 
 package org.eclipse.jetty.http;
 
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
 
 /**
- * HTTP compliance modes:
- * <dl>
- * <dt>RFC7230</dt><dd>(default) Compliance with RFC7230</dd>
- * <dt>RFC2616</dt><dd>Wrapped/Continued headers and HTTP/0.9 supported</dd>
- * <dt>LEGACY</dt><dd>(aka STRICT) Adherence to Servlet Specification requirement for 
- * exact case of header names, bypassing the header caches, which are case insensitive, 
- * otherwise equivalent to RFC2616</dd>
+ * HTTP compliance modes for Jetty HTTP parsing and handling.
+ * A Compliance mode consists of a set of {@link HttpComplianceSection}s which are applied
+ * when the mode is enabled.
+ * <p>
+ * Currently the set of modes is an enum and cannot be dynamically extended, but future major releases may convert this
+ * to a class. To modify modes there are four custom modes that can be modified by setting the property
+ * <code>org.eclipse.jetty.http.HttpCompliance.CUSTOMn</code> (where 'n' is '0', '1', '2' or '3'), to a comma separated
+ * list of sections.  The list should start with one of the following strings:<dl>
+ * <dt>0</dt><dd>No {@link HttpComplianceSection}s</dd>
+ * <dt>*</dt><dd>All {@link HttpComplianceSection}s</dd>
+ * <dt>RFC2616</dt><dd>The set of {@link HttpComplianceSection}s application to https://tools.ietf.org/html/rfc2616,
+ * but not https://tools.ietf.org/html/rfc7230</dd>
+ * <dt>RFC7230</dt><dd>The set of {@link HttpComplianceSection}s application to https://tools.ietf.org/html/rfc7230</dd>
  * </dl>
+ * The remainder of the list can contain then names of {@link HttpComplianceSection}s to include them in the mode, or prefixed
+ * with a '-' to exclude thm from the mode.    Note that Jetty's modes may have some historic minor differences from the strict
+ * RFC compliance, for example the <code>RFC2616</code> HttpCompliance is defines as 
+ * <code>RFC2616,-RFC7230_3_2_FIELD_COLON,-RFC7230_3_1_1_METHOD_CASE_SENSITIVE</code>.
+ * <p>
+ * Note also that the {@link EnumSet} return by {@link HttpCompliance#sections()} is mutable, so that modes may 
+ * be altered in code and will affect all usages of the mode.
  */
-public enum HttpCompliance { LEGACY, RFC2616, RFC7230 }
\ No newline at end of file
+public enum HttpCompliance // TODO in Jetty-10 convert this enum to a class so that extra custom modes can be defined dynamically
```

Comment:
Instead of a todo, how about just making an issue?

Reply:
I'll do both.  I'll create the issue IFF this PR ever gets accepted :)